### ISSUE TITLE
Cannot start kernel: "error in process sentinel", "Couldn't parse JSON"

### ISSUE THREAD

**Issue by c-vahl** [millejoh/emacs-ipython-notebook#568]
I'm running Emacs 26.2 from MinGW on Windows with EIN from Melpa (20190702.1230). I can log into my Jupyter server, but when I try opening a notebook, I get the following error in Emacs:

`error in process sentinel: Wrong type argument: number-or-marker-p, nil
E`

The notebook opens, but the kernel doesn't start. In Jupyter:

```
[W 13:17:12.295 NotebookApp] Notebook head/notebooks/Investigate_h5-File .ipynb is not trusted
[E 13:17:12.495 NotebookApp] Couldn't parse JSON
    Traceback (most recent call last):
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\site-packages\notebook\base\handlers.py", line 551, in get_json_body
        model = json.loads(body)
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\__init__.py", line 348, in loads
        return _default_decoder.decode(s)
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\decoder.py", line 337, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\decoder.py", line 355, in raw_decode
        raise JSONDecodeError("Expecting value", s, err.value) from None
    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0) [W 13:17:12.495 NotebookApp] 400 POST /api/sessions (127.0.0.1): Invalid JSON in body of request
```

Any help is appreciated. Let me know if you need more information.

**Comment by dickmao** [comment]
Have you tried opening the same notebook from the browser?

**Comment by c-vahl** [comment]
Yes, that does work. I get this erorr with all notebooks I've tried to open and they all work in the browser. I've also started a Jupyter server on another machine, conntected remotely and got the same error.

**Comment by dickmao** [comment]
As you are probably aware, the Jupyter server cannot parse `Investigate_h5-File .ipynb` (the space in the filename is perturbing).  It's not clear how it can suddenly successfully parse the same file when the client is the web browser instead of emacs.

I will minimally need the output of `ein:dev-bug-report-template` as described in the [README](https://github.com/millejoh/emacs-ipython-notebook/blob/master/README.rst).

**Comment by dickmao** [comment]
Ideally, you can supply the problematic `Investigate_h5-File .ipynb` file as well if it does not contain information sensitive to your institution.

**Comment by c-vahl** [comment]
I have tried with a new notebook I created in the browser (filename with no spaces), and the eror is the same. The log below is from the old file though.

## Problem description
I cannot start a kernel with EIN. In Emacs, I get

`error in process sentinel: Wrong type argument: number-or-marker-p, nil E`

In Jupyter server:

```
[W 13:17:12.295 NotebookApp] Notebook head/notebooks/Investigate_h5-File .ipynb is not trusted
[E 13:17:12.495 NotebookApp] Couldn't parse JSON
    Traceback (most recent call last):
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\site-packages\notebook\base\handlers.py", line 551, in get_json_body
        model = json.loads(body)
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\__init__.py", line 348, in loads
        return _default_decoder.decode(s)
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\decoder.py", line 337, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\decoder.py", line 355, in raw_decode
        raise JSONDecodeError("Expecting value", s, err.value) from None
    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0) [W 13:17:12.495 NotebookApp] 400 POST /api/sessions (127.0.0.1): Invalid JSON in body of request
```


## Steps to reproduce the problem
Login to a Jupyter notebook server and open a notebook.


## System info:

```cl
("EIN system info"
 :emacs-version "GNU Emacs 26.2 (build 1, x86_64-w64-mingw32)
 of 2019-06-03"
 :emacs-bzr-version "0e4951d953e629524553a4a265360984436fc3a1"
 :window-system w32
 :emacs-variant nil
 :os
 (:uname "MSYS_NT-10.0-17134 RAMSEY 3.0.7-338.x86_64 2019-07-03 08:42 UTC x86_64 Msys
"
         :lsb-release nil)
 :notebook
 "Name: notebook
Version: 5.7.8
Summary: A web-based notebook environment for interactive computing
Home-page: http://jupyter.org
Author: Jupyter Development Team
Author-email: jupyter@googlegroups.com
License: BSD
Location: c:\\users\\cvahl\\appdata\\local\\miniconda3\\envs\\py3\\lib\\site-packages
Requires: jinja2, ipython-genutils, tornado, jupyter-core, jupyter-client, nbconvert, terminado, prometheus-client, nbformat, Send2Trash, traitlets, pyzmq, ipykernel
Required-by: widgetsnbextension, jupyter, jupyter-nbextensions-configurator, jupyter-latex-envs, jupyter-contrib-nbextensions, jupyter-contrib-core
"
 :ipython "7.5.0
"
 :image-types
 (svg png gif tiff jpeg xpm xbm pbm)
 :image-types-available
 (png gif tiff jpeg xpm xbm pbm)
 :request
 (:backend curl)
 :ein
 (:version "20190702.1230"
           :source-dir "c:/Users/cvahl/.emacs.d/elpa/ein-20190702.1230/")
 :lib
 ((:name "websocket"
         :path "c:/Users/cvahl/.emacs.d/elpa/websocket-20190621.54/websocket.elc"
         :featurep t
         :version-var websocket-version
         :version "1.11.1")
  (:name "auto-complete"
         :path "c:/Users/cvahl/.emacs.d/elpa/auto-complete-20170125.245/auto-complete.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "request"
         :path "c:/Users/cvahl/.emacs.d/elpa/request-20190621.1622/request.elc"
         :featurep t
         :version-var request-version
         :version "0.3.0")
  (:name "deferred"
         :path "c:/Users/cvahl/.emacs.d/elpa/deferred-20170901.1330/deferred.elc"
         :featurep t
         :version-var deferred:version
         :version "0.5.0")
  (:name "polymode"
         :path "c:/Users/cvahl/.emacs.d/elpa/polymode-20190624.1927/polymode.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "markdown-mode"
         :path "c:/Users/cvahl/.emacs.d/elpa/markdown-mode-20190305.319/markdown-mode.elc"
         :featurep nil
         :version-var nil
         :version nil)
  (:name "dash"
         :path "c:/Users/cvahl/.emacs.d/elpa/dash-20190424.1804/dash.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "s"
         :path "c:/Users/cvahl/.emacs.d/elpa/s-20180406.808/s.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "skewer-mode"
         :path "c:/Users/cvahl/.emacs.d/elpa/skewer-mode-20180706.1807/skewer-mode.elc"
         :featurep t
         :version-var nil
         :version nil)))
```

**Comment by millejoh** [comment]
Can you create a new notebook using ein? For the sake of completeness it would be good to see the ipynb file of the empty notebook you created from the browser.

**Comment by c-vahl** [comment]
When I try to create a new notebook in EIN, Emacs gives:

`ein: [error] ein:notebook-list-new-notebook-error: error (error http 400)`

And Jupyter server:

```
[E 14:25:16.341 NotebookApp] Couldn't parse JSON
    Traceback (most recent call last):
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\site-packages\notebook\base\handlers.py", line 551, in get_json_body
        model = json.loads(body)
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\__init__.py", line 348, in loads
        return _default_decoder.decode(s)
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\decoder.py", line 337, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\decoder.py", line 355, in raw_decode
        raise JSONDecodeError("Expecting value", s, err.value) from None
    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0) [W 14:25:16.341 NotebookApp] 400 POST /api/contents (127.0.0.1): Invalid JSON in body of request
[W 14:25:16.341 NotebookApp] Invalid JSON in body of request
[W 14:25:16.341 NotebookApp] 400 POST /api/contents (127.0.0.1) 0.00ms referer=None
[E 14:25:18.014 NotebookApp] Couldn't parse JSON
    Traceback (most recent call last):
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\site-packages\notebook\base\handlers.py", line 551, in get_json_body
        model = json.loads(body)
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\__init__.py", line 348, in loads
        return _default_decoder.decode(s)
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\decoder.py", line 337, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "c:\users\cvahl\appdata\local\miniconda3\envs\py3\lib\json\decoder.py", line 355, in raw_decode
        raise JSONDecodeError("Expecting value", s, err.value) from None
    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0) [W 14:25:18.014 NotebookApp] 400 POST /api/contents (127.0.0.1): Invalid JSON in body of request
[W 14:25:18.014 NotebookApp] Invalid JSON in body of request
[W 14:25:18.030 NotebookApp] 400 POST /api/contents (127.0.0.1) 15.62ms referer=None
```

Maybe there is a way to look into the invalid JSON?

[empty_notebook.zip](https://github.com/millejoh/emacs-ipython-notebook/files/3372935/empty_notebook.zip)

**Comment by millejoh** [comment]
Thank you. I’m looking at all this from my cell phone, so I don’t have much in the way of advice at the moment. I will try to look at this in more depth later today. Do you know which curl executable emacs is configured to use? By that I mean the path to the executable and its version. If you have another, more recent version of curl on your machine you might try using that.

**Comment by c-vahl** [comment]
It"s not very urgent, thanks for looking into this.

curl should be up to date:
```
curl 7.65.1 (x86_64-w64-mingw32) libcurl/7.65.1 OpenSSL/1.1.1c (Schannel) zlib/1.2.11 brotli/1.0.7 libidn2/2.2.0 libpsl/0.21.0 (+libidn2/2.1.1) nghttp2/1.39.1
Release-Date: 2019-06-05
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smtp smtps telnet tftp 
Features: AsynchDNS brotli HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz Metalink MultiSSL NTLM PSL SPNEGO SSL SSPI TLS-SRP
```

It is located at `c:/msys64/usr/bin`.

**Comment by millejoh** [comment]
Thank you for the notebook file. I was able to open it on my system no problem. To eliminate curl as the issue can you set `request-curl` to the absolute path to your curl (i.e. `c:/msys64/usr/bin/curl.exe`)?

Also, how are you starting jupyter? Have you set any config parameters different from their default?

**Comment by c-vahl** [comment]
I have set `request-curl` as you asked with no change.

I have also tried EIN with a clean Jupyter (from Conda) with the same result.

**Comment by millejoh** [comment]
Okay, it was worth a try. I am unfortunately unable to reproduce the error. I am tempted to suggest that you try running emacs+ein from the WSL, but if you don't mind doing some additional work there is something to try to get more information.

First you need to `M-x ein:dev-start-debug` and set both `request-log-level` and `request-message-level` to `debug`. Then do what you need to do to reproduce the error and dump the contents from `*Message*` buffer and the websocket log buffer (the name will start with `*websocket`) back here. If we are lucky those messages will provide further insight into why Jupyter is generating the 400 response.

**Comment by c-vahl** [comment]
Unfortunately, I can't use WSL at my workplace.

Where exactly do I set `request-log-level` and `request-message-level`?

**Comment by dickmao** [comment]
The error tracebacks from c-vahl all point to empty or malformed notebook API requests, not kernel websocket messages.  And `request-log-level` and `request-message-level` are already set appropriately by the README instructions.

For whatever reason, the `json-encode` of his POST requests are malformed or more likely empty.  `M-x ein:log-pop-to-request-buffer` is key.  I am generally opposed addressing Windows issues since their experience will still be poor (because of more unforeseen incompatibilities) after resolving the typical initial "just-get-it-running" issues.

**Comment by millejoh** [comment]
@c-vahl - I am stuck in my old ways, @dickmao is correct that all you need to do is call `M-x ein:dev-start-debug` then follow up with calls to `M-x ein:log-pop-to-all-buffer` to see ein's log and `M-x ein:log-pop-to-request-buffer`. The second will probably be of more interest for this exercise.

My company-provided Windows laptop can be sensitive when it comes to the ein/jupyter communication. I have had to disable token authentication (`c.NotebookApp.token = ''`) and be more permissive on origin access (`c.NotebookApp.allow_origin = '*'`) to get the combination working reliably.

I realize not all of us have the luxury of choosing our computing platform, so I'll do what I can to help you troubleshoot this. In the end the issue may be with how your IT group has configured your machine, which is obviously something we cannot control, but maybe can at least work around.

**Comment by c-vahl** [comment]
I very much appreciate your help. I've added the settings you suggested to Jupyter.

Here are the logs when trying to open a newly created notebook (keep in mind that the notebook opens, only the kernel does not start):

*ein:log-all*
```
10:33:33:405: [verbose] Skipping login http://127.0.0.1:8888 @#<buffer *ein:log-all*>
10:33:33:511: [debug] ein:query-notebook-version--complete STATUS: 200 DATA: (:version 5.7.8) @#<buffer *ein:log-all*>
10:33:33:638: [debug] ein:query-kernelspecs--complete STATUS: 200 DATA: (:default python3 :kernelspecs (:python3 (:name python3 :spec (:argv (python -m ipykernel_launcher -f {connection_file}) :env nil :display_name Python 3 :language python :interrupt_mode signal :metadata nil) :resources (:logo-32x32 /kernelspecs/python3/logo-32x32.png :logo-64x64 /kernelspecs/python3/logo-64x64.png)))) @#<buffer *ein:log-all*>
10:33:33:787: [debug] Race! (content-query-sessions http://127.0.0.1:8888) nil @#<buffer *ein:notebooklist http://127.0.0.1:8888*>
10:33:33:793: [verbose] Opened notebooklist at http://127.0.0.1:8888 @#<buffer *ein:notebooklist http://127.0.0.1:8888*>
10:33:33:794: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name  :path  :last_modified 2019-07-10T13:02:47.067856Z :created 2018-09-26T14:54:23.008146Z :content ((:name evolopro :path evolopro :last_modified 2019-06-04T14:48:29.132975Z :created 2019-06-04T14:47:45.672726Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name Flex4Apps :path Flex4Apps :last_modified 2019-07-11T08:19:53.020244Z :created 2018-12-03T12:30:49.253978Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name Flex4Apps_back :path Flex4Apps_back :last_modified 2018-10-22T13:08:38Z :created 2018-10-22T13:08:38.655594Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name Flex4Apps_back_2 :path Flex4Apps_back_2 :last_modified 2018-10-22T13:49:14Z :created 2018-10-22T13:49:14.116956Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name head :path head :last_modified 2019-07-10T14:33 @#<buffer *ein:log-all*>
10:33:33:847: [debug] ein:query-sessions--complete STATUS: 200 DATA: nil @#<buffer *ein:log-all*>
10:33:33:987: [debug] ein:query-sessions--complete STATUS: 200 DATA: nil @#<buffer *ein:log-all*>
10:33:34:037: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name  :path  :last_modified 2019-07-10T13:02:47.067856Z :created 2018-09-26T14:54:23.008146Z :content ((:name evolopro :path evolopro :last_modified 2019-06-04T14:48:29.132975Z :created 2019-06-04T14:47:45.672726Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name Flex4Apps :path Flex4Apps :last_modified 2019-07-11T08:19:53.020244Z :created 2018-12-03T12:30:49.253978Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name Flex4Apps_back :path Flex4Apps_back :last_modified 2018-10-22T13:08:38Z :created 2018-10-22T13:08:38.655594Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name Flex4Apps_back_2 :path Flex4Apps_back_2 :last_modified 2018-10-22T13:49:14Z :created 2018-10-22T13:49:14.116956Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name head :path head :last_modified 2019-07-10T14:33 @#<buffer *ein:log-all*>
10:33:34:227: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name evolopro :path evolopro :last_modified 2019-06-04T14:48:29.132975Z :created 2019-06-04T14:47:45.672726Z :content ((:name adaweld :path evolopro/adaweld :last_modified 2019-06-04T14:56:21.024272Z :created 2019-06-04T14:48:29.132975Z :content nil :format nil :mimetype nil :size nil :writable t :type directory)) :format json :mimetype nil :size nil :writable t :type directory) @#<buffer *ein:log-all*>
10:33:34:311: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name head :path head :last_modified 2019-07-10T14:33:06.651973Z :created 2019-07-01T13:51:48.612173Z :content ((:name head :path head/head :last_modified 2019-07-01T13:51:55.877018Z :created 2019-07-01T13:51:55.754035Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name lrp :path head/lrp :last_modified 2019-07-01T14:24:14.785481Z :created 2019-07-01T13:51:55.890115Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name LRP.py :path head/LRP.py :last_modified 2019-07-10T11:21:08.804010Z :created 2019-07-01T13:51:55.676408Z :content nil :format nil :mimetype text/plain :size 3522 :writable t :type file) (:name LRP_debug.py :path head/LRP_debug.py :last_modified 2019-07-10T14:33:06.625743Z :created 2019-07-10T11:30:38.944229Z :content nil :format nil :mimetype text/plain :size 6010 :writable t :type file) (:name LSTM-description.aux :path head/LSTM-descri @#<buffer *ein:log-all*>
10:33:34:317: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name Flex4Apps_back :path Flex4Apps_back :last_modified 2018-10-22T13:08:38Z :created 2018-10-22T13:08:38.655594Z :content ((:name Flex4Apps :path Flex4Apps_back/Flex4Apps :last_modified 2019-06-04T16:28:40.335527Z :created 2018-10-22T13:17:12.040798Z :content nil :format nil :mimetype nil :size nil :writable t :type directory)) :format json :mimetype nil :size nil :writable t :type directory) @#<buffer *ein:log-all*>
10:33:34:355: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name head - Copy :path head - Copy :last_modified 2019-07-08T08:25:27.756871Z :created 2019-07-08T08:25:25.149886Z :content ((:name head :path head - Copy/head :last_modified 2019-07-08T08:25:26.621411Z :created 2019-07-08T08:25:26.512427Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name lrp :path head - Copy/lrp :last_modified 2019-07-08T08:25:26.830158Z :created 2019-07-08T08:25:26.627465Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name LRP.py :path head - Copy/LRP.py :last_modified 2019-07-03T17:00:05.434960Z :created 2019-07-08T08:25:25.206359Z :content nil :format nil :mimetype text/plain :size 3383 :writable t :type file) (:name LSTM-description.aux :path head - Copy/LSTM-description.aux :last_modified 2019-07-01T13:57:35.635256Z :created 2019-07-08T08:25:25.229556Z :content nil :format nil :mimetype nil :size 713 :writable t :type file)  @#<buffer *ein:log-all*>
10:33:34:361: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name Flex4Apps_back_2 :path Flex4Apps_back_2 :last_modified 2018-10-22T13:49:14Z :created 2018-10-22T13:49:14.116956Z :content ((:name Flex4Apps :path Flex4Apps_back_2/Flex4Apps :last_modified 2019-06-04T16:28:58.991751Z :created 2018-10-22T13:53:22.459939Z :content nil :format nil :mimetype nil :size nil :writable t :type directory)) :format json :mimetype nil :size nil :writable t :type directory) @#<buffer *ein:log-all*>
10:33:34:496: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name adaweld :path evolopro/adaweld :last_modified 2019-06-04T14:56:21.024272Z :created 2019-06-04T14:48:29.132975Z :content ((:name notebooks :path evolopro/adaweld/notebooks :last_modified 2019-06-04T15:54:40.197252Z :created 2019-06-04T14:56:21.024272Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name README.md :path evolopro/adaweld/README.md :last_modified 2019-06-04T14:48:34.368077Z :created 2019-06-04T14:48:34.367068Z :content nil :format nil :mimetype nil :size 457 :writable t :type file)) :format json :mimetype nil :size nil :writable t :type directory) @#<buffer *ein:log-all*>
10:33:34:564: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name Flex4Apps :path Flex4Apps :last_modified 2019-07-11T08:19:53.020244Z :created 2018-12-03T12:30:49.253978Z :content ((:name build :path Flex4Apps/build :last_modified 2019-06-28T09:06:40.724974Z :created 2019-06-25T14:58:08.294239Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name CMakeLists.txt :path Flex4Apps/CMakeLists.txt :last_modified 2018-12-05T12:16:02.955920Z :created 2018-12-05T12:16:02.947850Z :content nil :format nil :mimetype text/plain :size 603 :writable t :type file) (:name docs :path Flex4Apps/docs :last_modified 2019-05-21T12:06:37.788774Z :created 2018-12-03T12:30:57.481425Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name external :path Flex4Apps/external :last_modified 2018-12-03T12:30:57.903271Z :created 2018-12-03T12:30:57.902262Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name f4a : @#<buffer *ein:log-all*>
10:33:34:622: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name lrp :path head/lrp :last_modified 2019-07-01T14:24:14.785481Z :created 2019-07-01T13:51:55.890115Z :content ((:name Activations.py :path head/lrp/Activations.py :last_modified 2019-07-01T13:51:55.937491Z :created 2019-07-01T13:51:55.918343Z :content nil :format nil :mimetype text/plain :size 145 :writable t :type file) (:name Dense.py :path head/lrp/Dense.py :last_modified 2019-07-01T13:51:55.951625Z :created 2019-07-01T13:51:55.950611Z :content nil :format nil :mimetype text/plain :size 1350 :writable t :type file) (:name lrp.aux :path head/lrp/lrp.aux :last_modified 2019-07-01T13:57:11.848422Z :created 2019-07-01T13:53:08.162396Z :content nil :format nil :mimetype nil :size 9 :writable t :type file) (:name lrp.log :path head/lrp/lrp.log :last_modified 2019-07-01T13:57:11.854476Z :created 2019-07-01T13:52:56.541020Z :content nil :format nil :mimetype nil :size 12908 :writable t :type file) (:name lrp.pdf :path @#<buffer *ein:log-all*>
10:33:34:623: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name head :path head/head :last_modified 2019-07-01T13:51:55.877018Z :created 2019-07-01T13:51:55.754035Z :content ((:name load_and_predict.py :path head/head/load_and_predict.py :last_modified 2019-07-01T13:51:55.824622Z :created 2019-07-01T13:51:55.813535Z :content nil :format nil :mimetype text/plain :size 1688 :writable t :type file) (:name plotting.py :path head/head/plotting.py :last_modified 2019-07-01T13:51:55.862936Z :created 2019-07-01T13:51:55.838749Z :content nil :format nil :mimetype text/plain :size 2911 :writable t :type file) (:name util.py :path head/head/util.py :last_modified 2019-07-01T13:51:55.879029Z :created 2019-07-01T13:51:55.877018Z :content nil :format nil :mimetype text/plain :size 746 :writable t :type file) (:name __init__.py :path head/head/__init__.py :last_modified 2019-07-01T13:51:55.799413Z :created 2019-07-01T13:51:55.799413Z :content nil :format nil :mimetype text/plain :size 0 : @#<buffer *ein:log-all*>
10:33:34:765: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name Flex4Apps :path Flex4Apps_back_2/Flex4Apps :last_modified 2019-06-04T16:28:58.991751Z :created 2018-10-22T13:53:22.459939Z :content ((:name build :path Flex4Apps_back_2/Flex4Apps/build :last_modified 2018-10-02T16:44:06Z :created 2018-10-22T13:53:40.578304Z :content nil :format nil :mimetype nil :size nil :writable t :type directory)) :format json :mimetype nil :size nil :writable t :type directory) @#<buffer *ein:log-all*>
10:33:34:832: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name py :path head - Copy/py :last_modified 2019-07-08T08:25:27.774030Z :created 2019-07-08T08:25:27.756871Z :content ((:name classification.py :path head - Copy/py/classification.py :last_modified 2019-07-08T08:24:16.932158Z :created 2019-07-08T08:25:27.774030Z :content nil :format nil :mimetype text/plain :size 4273 :writable t :type file)) :format json :mimetype nil :size nil :writable t :type directory) @#<buffer *ein:log-all*>
10:33:34:835: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name notebooks :path head - Copy/notebooks :last_modified 2019-07-08T08:25:27.717504Z :created 2019-07-08T08:25:27.038885Z :content ((:name explorations.ipynb :path head - Copy/notebooks/explorations.ipynb :last_modified 2019-07-02T12:55:10.483002Z :created 2019-07-08T08:25:27.044943Z :content nil :format nil :mimetype nil :size 166767 :writable t :type notebook) (:name explorations.py :path head - Copy/notebooks/explorations.py :last_modified 2019-07-03T15:08:24.400329Z :created 2019-07-08T08:25:27.092341Z :content nil :format nil :mimetype text/plain :size 4926 :writable t :type file) (:name explorations_leo.py :path head - Copy/notebooks/explorations_leo.py :last_modified 2019-07-02T13:40:43.176801Z :created 2019-07-08T08:25:27.106464Z :content nil :format nil :mimetype text/plain :size 4842 :writable t :type file) (:name Investigate_h5-File .ipynb :path head - Copy/notebooks/Investigate_h5-File .ipynb :last_modi @#<buffer *ein:log-all*>
10:33:34:836: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name lrp :path head - Copy/lrp :last_modified 2019-07-08T08:25:26.830158Z :created 2019-07-08T08:25:26.627465Z :content ((:name Activations.py :path head - Copy/lrp/Activations.py :last_modified 2019-07-01T13:51:55.937491Z :created 2019-07-08T08:25:26.639580Z :content nil :format nil :mimetype text/plain :size 145 :writable t :type file) (:name Dense.py :path head - Copy/lrp/Dense.py :last_modified 2019-07-01T13:51:55.951625Z :created 2019-07-08T08:25:26.660756Z :content nil :format nil :mimetype text/plain :size 1350 :writable t :type file) (:name lrp.aux :path head - Copy/lrp/lrp.aux :last_modified 2019-07-01T13:57:11.848422Z :created 2019-07-08T08:25:26.681930Z :content nil :format nil :mimetype nil :size 9 :writable t :type file) (:name lrp.log :path head - Copy/lrp/lrp.log :last_modified 2019-07-01T13:57:11.854476Z :created 2019-07-08T08:25:26.709168Z :content nil :format nil :mimetype nil :size 12908 :writable @#<buffer *ein:log-all*>
10:33:34:837: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name head :path head - Copy/head :last_modified 2019-07-08T08:25:26.621411Z :created 2019-07-08T08:25:26.512427Z :content ((:name load_and_predict.py :path head - Copy/head/load_and_predict.py :last_modified 2019-07-01T13:51:55.824622Z :created 2019-07-08T08:25:26.528580Z :content nil :format nil :mimetype text/plain :size 1688 :writable t :type file) (:name plotting.py :path head - Copy/head/plotting.py :last_modified 2019-07-01T13:51:55.862936Z :created 2019-07-08T08:25:26.538674Z :content nil :format nil :mimetype text/plain :size 2911 :writable t :type file) (:name util.py :path head - Copy/head/util.py :last_modified 2019-07-01T13:51:55.879029Z :created 2019-07-08T08:25:26.585074Z :content nil :format nil :mimetype text/plain :size 746 :writable t :type file) (:name __init__.py :path head - Copy/head/__init__.py :last_modified 2019-07-01T13:51:55.799413Z :created 2019-07-08T08:25:26.621411Z :content nil :format @#<buffer *ein:log-all*>
10:33:34:839: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name Flex4Apps :path Flex4Apps_back/Flex4Apps :last_modified 2019-06-04T16:28:40.335527Z :created 2018-10-22T13:17:12.040798Z :content ((:name build :path Flex4Apps_back/Flex4Apps/build :last_modified 2018-10-02T16:44:06Z :created 2018-10-22T13:17:19.796230Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name docs :path Flex4Apps_back/Flex4Apps/docs :last_modified 2018-10-18T13:53:28Z :created 2018-10-22T13:17:20.412959Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name f4a :path Flex4Apps_back/Flex4Apps/f4a :last_modified 2019-06-04T16:28:35.058384Z :created 2018-10-22T13:17:21.091239Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name LegacyImplementation :path Flex4Apps_back/Flex4Apps/LegacyImplementation :last_modified 2018-10-18T13:53:28Z :created 2018-10-22T13:17:19.096271Z :content nil :format nil :mimetype n @#<buffer *ein:log-all*>
10:33:34:840: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name py :path head/py :last_modified 2019-07-08T09:41:48.056722Z :created 2019-07-03T11:16:38.441593Z :content ((:name classification.py :path head/py/classification.py :last_modified 2019-07-08T09:41:48.046630Z :created 2019-07-03T11:16:38.482946Z :content nil :format nil :mimetype text/plain :size 4261 :writable t :type file)) :format json :mimetype nil :size nil :writable t :type directory) @#<buffer *ein:log-all*>
10:33:34:844: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name notebooks :path head/notebooks :last_modified 2019-07-10T14:05:20.349708Z :created 2019-07-01T13:51:56.022233Z :content ((:name explorations.ipynb :path head/notebooks/explorations.ipynb :last_modified 2019-07-02T12:55:10.483002Z :created 2019-07-01T15:16:10.986894Z :content nil :format nil :mimetype nil :size 166767 :writable t :type notebook) (:name explorations.py :path head/notebooks/explorations.py :last_modified 2019-07-09T11:11:21.639380Z :created 2019-07-01T15:16:11.050435Z :content nil :format nil :mimetype text/plain :size 4926 :writable t :type file) (:name explorations_leo.py :path head/notebooks/explorations_leo.py :last_modified 2019-07-02T13:40:43.176801Z :created 2019-07-02T12:36:33.167093Z :content nil :format nil :mimetype text/plain :size 4842 :writable t :type file) (:name Investigate_h5-File .ipynb :path head/notebooks/Investigate_h5-File .ipynb :last_modified 2019-07-01T13:51:56.045424Z :c @#<buffer *ein:log-all*>
10:33:34:888: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name external :path Flex4Apps/external :last_modified 2018-12-03T12:30:57.903271Z :created 2018-12-03T12:30:57.902262Z :content ((:name logparser :path Flex4Apps/external/logparser :last_modified 2018-12-03T12:30:58.997118Z :created 2018-12-03T12:30:57.903271Z :content nil :format nil :mimetype nil :size nil :writable t :type directory)) :format json :mimetype nil :size nil :writable t :type directory) @#<buffer *ein:log-all*>
10:33:34:919: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name LegacyImplementation :path Flex4Apps/LegacyImplementation :last_modified 2018-12-03T12:30:57.468307Z :created 2018-12-03T12:30:57.283631Z :content ((:name CMakeLists.txt :path Flex4Apps/LegacyImplementation/CMakeLists.txt :last_modified 2018-12-03T12:30:57.295742Z :created 2018-12-03T12:30:57.284641Z :content nil :format nil :mimetype text/plain :size 596 :writable t :type file) (:name Demonstrator :path Flex4Apps/LegacyImplementation/Demonstrator :last_modified 2018-12-03T12:30:57.390600Z :created 2018-12-03T12:30:57.295742Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name ElasticSearchConnection :path Flex4Apps/LegacyImplementation/ElasticSearchConnection :last_modified 2018-12-03T12:30:57.398675Z :created 2018-12-03T12:30:57.391604Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name IntegratedTool_V1 :path Flex4Apps/LegacyImplementation/In @#<buffer *ein:log-all*>
10:33:34:932: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name docs :path Flex4Apps/docs :last_modified 2019-05-21T12:06:37.788774Z :created 2018-12-03T12:30:57.481425Z :content ((:name 15025_Flex4Apps_Full_Project_Proposal.pdf :path Flex4Apps/docs/15025_Flex4Apps_Full_Project_Proposal.pdf :last_modified 2018-12-03T12:30:57.509697Z :created 2018-12-03T12:30:57.483443Z :content nil :format nil :mimetype application/pdf :size 2798857 :writable t :type file) (:name Conference Call Slides :path Flex4Apps/docs/Conference Call Slides :last_modified 2018-12-03T12:30:57.540979Z :created 2018-12-03T12:30:57.528861Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name Consortium Meeting Slides :path Flex4Apps/docs/Consortium Meeting Slides :last_modified 2018-12-03T12:30:57.569243Z :created 2018-12-03T12:30:57.561169Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name CPS_HUB_NRW :path Flex4Apps/docs/CPS_HUB_NRW :last @#<buffer *ein:log-all*>
10:33:34:951: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name build :path Flex4Apps/build :last_modified 2019-06-28T09:06:40.724974Z :created 2019-06-25T14:58:08.294239Z :content ((:name build.ninja :path Flex4Apps/build/build.ninja :last_modified 2019-06-28T09:06:40.713874Z :created 2019-06-28T09:06:40.687643Z :content nil :format nil :mimetype nil :size 40865 :writable t :type file) (:name CMakeCache.txt :path Flex4Apps/build/CMakeCache.txt :last_modified 2019-06-28T09:03:28.591508Z :created 2019-06-28T09:03:28.564198Z :content nil :format nil :mimetype text/plain :size 14258 :writable t :type file) (:name CMakeFiles :path Flex4Apps/build/CMakeFiles :last_modified 2019-06-28T09:06:40.707822Z :created 2019-06-25T14:58:12.666554Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name cmake_install.cmake :path Flex4Apps/build/cmake_install.cmake :last_modified 2019-06-25T14:58:16.443535Z :created 2019-06-25T14:58:16.442526Z :content nil :forma @#<buffer *ein:log-all*>
10:33:34:979: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name f4a :path Flex4Apps/f4a :last_modified 2018-12-03T12:32:41.660591Z :created 2018-12-03T12:30:57.667171Z :content ((:name clustering :path Flex4Apps/f4a/clustering :last_modified 2018-12-03T12:30:57.673223Z :created 2018-12-03T12:30:57.668180Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name logline :path Flex4Apps/f4a/logline :last_modified 2018-12-03T12:30:57.679276Z :created 2018-12-03T12:30:57.674232Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name parameter_filter :path Flex4Apps/f4a/parameter_filter :last_modified 2019-05-21T12:07:29.422349Z :created 2018-12-03T12:30:57.681292Z :content nil :format nil :mimetype nil :size nil :writable t :type directory) (:name string :path Flex4Apps/f4a/string :last_modified 2019-05-21T12:07:29.473809Z :created 2018-12-03T12:30:57.689364Z :content nil :format nil :mimetype nil :size nil :writable t : @#<buffer *ein:log-all*>
10:33:35:026: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name jupyter :path Flex4Apps/jupyter :last_modified 2019-07-05T10:04:59.253328Z :created 2019-05-21T11:53:48.268953Z :content ((:name layout_settings.py :path Flex4Apps/jupyter/layout_settings.py :last_modified 2019-06-11T09:31:31.108693Z :created 2019-06-11T09:31:31.107683Z :content nil :format nil :mimetype text/plain :size 704 :writable t :type file) (:name nokia_log_classification.ipynb :path Flex4Apps/jupyter/nokia_log_classification.ipynb :last_modified 2019-06-28T13:44:08.488374Z :created 2019-06-11T09:31:31.167196Z :content nil :format nil :mimetype nil :size 198984 :writable t :type notebook) (:name nokia_log_classification.py :path Flex4Apps/jupyter/nokia_log_classification.py :last_modified 2019-06-28T13:44:08.613389Z :created 2019-06-11T09:31:31.216640Z :content nil :format nil :mimetype text/plain :size 21675 :writable t :type file) (:name nokia_log_exploration.ipynb :path Flex4Apps/jupyter/nokia_log_ex @#<buffer *ein:log-all*>
10:34:00:170: [verbose] Opened notebook Untitled.ipynb @#<buffer *ein:notebooklist http://127.0.0.1:8888*>
10:34:00:170: [debug] [GC-PREPARE-OPERATION] Setting cons threshold to 8000000000. @#<buffer *ein:notebooklist http://127.0.0.1:8888*>
10:34:00:172: [info] Worksheet Untitled.ipynb is ready @#<buffer *ein: http://127.0.0.1:8888/Untitled.ipynb*>
10:34:00:188: [verbose] Enabling autosaves for Untitled.ipynb with frequency 300 seconds. @#<buffer *ein:notebooklist http://127.0.0.1:8888*>
10:34:00:194: [debug] [GC-COMPLETE-OPERATION] Reverting cons threshold to 800000. @#<buffer *ein:notebooklist http://127.0.0.1:8888*>
10:34:00:194: [debug] ein:query-contents--complete STATUS: 200 DATA: (:name Untitled.ipynb :path Untitled.ipynb :last_modified 2019-07-09T11:52:55.884299Z :created 2019-07-09T11:50:54.455303Z :content (:cells ((:cell_type markdown :metadata (:toc t) :source <h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"></ul></div>) (:cell_type code :execution_count 1 :metadata (:trusted t) :outputs ((:name stdout :output_type stream :text hey
)) :source print('hey')) (:cell_type code :execution_count nil :metadata (:trusted t) :outputs nil :source )) :metadata (:jupytext (:text_representation (:extension .py :format_name light :format_version 1.4 :jupytext_version 1.1.3)) :kernelspec (:display_name Python 3 :language python :name python3) :language_info (:codemirror_mode (:name ipython :version 3) :file_extension .py :mimetype text/x-python :name python :nbconvert_exporter python :pygments_lexer ipython3 :version 3.7.3) :toc (:base_numbering 1 :nav_menu n @#<buffer *ein:notebooklist http://127.0.0.1:8888*>
```

*Messages*
```
debug-on-error=t websocket-debug=t websocket-callback-debug-on-error=nil ein:debug=t ein:log-level=debug ein:log-message-level=verbose
ein: [verbose] Skipping login http://127.0.0.1:8888
ein: [verbose] Opened notebooklist at http://127.0.0.1:8888
ein: [verbose] Opened notebook Untitled.ipynb
ein: [info] Worksheet Untitled.ipynb is ready
ein: [verbose] Enabling autosaves for Untitled.ipynb with frequency 300 seconds.
Entering debugger...
```

*Backtrace*
```
Debugger entered--Lisp error: (wrong-type-argument number-or-marker-p nil)
  #f(compiled-function (proc event) #<bytecode 0x2e44401>)(#<process request curl> "finished\n")
  apply(#f(compiled-function (proc event) #<bytecode 0x2e44401>) (#<process request curl> "finished\n"))
  request--curl-callback(#<process request curl> "finished\n")
```

*request-log*
```
[debug] REQUEST--CURL-CALLBACK proc = #<process request curl>
[debug] REQUEST--CURL-CALLBACK buffer = #<buffer  *request curl*>
[debug] REQUEST--CURL-CALLBACK symbol-status = nil
[debug] REQUEST--CALLBACK
[debug] (buffer-string) =
HTTP/1.1 200 OK

Server: TornadoServer/6.0.3

Content-Type: application/json

Date: Thu, 11 Jul 2019 08:34:00 GMT

X-Content-Type-Options: nosniff

Content-Security-Policy: frame-ancestors 'self'; report-uri /api/security/csp-report; default-src 'none'

Access-Control-Allow-Origin: *

Last-Modified: Tue, 09 Jul 2019 11:52:55 GMT

Etag: "e2eba2f289b9929372de843c44802da96f73ada6"

Content-Length: 1489



{"name": "Untitled.ipynb", "path": "Untitled.ipynb", "last_modified": "2019-07-09T11:52:55.884299Z", "created": "2019-07-09T11:50:54.455303Z", "content": {"cells": [{"cell_type": "markdown", "metadata": {"toc": true}, "source": "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n<div class=\"toc\"><ul class=\"toc-item\"></ul></div>"}, {"cell_type": "code", "execution_count": 1, "metadata": {"trusted": true}, "outputs": [{"name": "stdout", "output_type": "stream", "text": "hey\n"}], "source": "print('hey')"}, {"cell_type": "code", "execution_count": null, "metadata": {"trusted": true}, "outputs": [], "source": ""}], "metadata": {"jupytext": {"text_representation": {"extension": ".py", "format_name": "light", "format_version": "1.4", "jupytext_version": "1.1.3"}}, "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.7.3"}, "toc": {"base_numbering": 1, "nav_menu": {}, "number_sections": true, "sideBar": true, "skip_h1_title": true, "title_cell": "Table of Contents", "title_sidebar": "Contents", "toc_cell": true, "toc_position": {}, "toc_section_display": true, "toc_window_display": false}}, "nbformat": 4, "nbformat_minor": 2}, "format": "json", "mimetype": null, "size": 1494, "writable": true, "type": "notebook"}
[debug] REQUEST-RESPONSE--CANCEL-TIMER
[debug] -CLEAN-HEADER
[debug] -CUT-HEADER
[debug] error-thrown = nil
[debug] -PARSE-DATA
[debug] parser = ein:json-read
[debug] data = (:name Untitled.ipynb :path Untitled.ipynb :last_modified 2019-07-09T11:52:55.884299Z :created 2019-07-09T11:50:54.455303Z :content (:cells ((:cell_type markdown :metadata (:toc t) :source <h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"></ul></div>) (:cell_type code :execution_count 1 :metadata (:trusted t) :outputs ((:name stdout :output_type stream :text hey
)) :source print('hey')) (:cell_type code :execution_count nil :metadata (:trusted t) :outputs nil :source )) :metadata (:jupytext (:text_representation (:extension .py :format_name light :format_version 1.4 :jupytext_version 1.1.3)) :kernelspec (:display_name Python 3 :language python :name python3) :language_info (:codemirror_mode (:name ipython :version 3) :file_extension .py :mimetype text/x-python :name python :nbconvert_exporter python :pygments_lexer ipython3 :version 3.7.3) :toc (:base_numbering 1 :nav_menu nil :number_sections t :sideBar t :skip_h1_title t :title_cell Table of Contents :title_sidebar Contents :toc_cell t :toc_position nil :toc_section_display t :toc_window_display :json-false)) :nbformat 4 :nbformat_minor 2) :format json :mimetype nil :size 1494 :writable t :type notebook)
[debug] symbol-status = success
[debug] Executing success callback.
[debug] REQUEST
[debug] Run: curl --silent --include --location --compressed --cookie c:/Users/cvahl/.emacs.d/request/curl-cookie-jar --cookie-jar c:/Users/cvahl/.emacs.d/request/curl-cookie-jar --write-out \n(:num-redirects %{num_redirects} :url-effective %{url_effective}) --data-binary @c:/Users/cvahl/AppData/Local/Temp/emacs-request8ikQWR --request POST --header X-XSRFTOKEN: 2|ac6ff301|9f2e547f5a8d24538b405769c644b3de|1562674358 http://127.0.0.1:8888/api/sessions
[debug] Executing complete callback.
[debug] c:/Users/cvahl/AppData/Local/Temp/curl-trace unreadable
[debug] REQUEST--CURL-CALLBACK event = finished

[debug] REQUEST--CURL-CALLBACK proc = #<process request curl>
[debug] REQUEST--CURL-CALLBACK buffer = #<buffer  *request curl*>
[debug] REQUEST--CURL-CALLBACK symbol-status = nil
```

**Comment by millejoh** [comment]
I feel like something is missing here - everything seems to go okay up until ein tries to establish a kernel session, but at that point all the log messages stop. I've had similar issues, but instead of getting malformed json errors I get 403 forbidden responses complaining the xsrf token is missing.

This is a total stab in the dark, but if you are running jupyter locally and aren't concerned about potential malicious attacks on your machine could you try running jupyter with xsrf authentication disabled. Do this by setting `c.NotebookApp.disable_check_xsrf = True` in your local jupyter notebook config file.

**Comment by c-vahl** [comment]
I've tried without xsrf authentication but get the same error as above. For now, I'm using Emacs on a Linux machine via an X-server. But if you have any other ideas I could tryout, let me know.

**Comment by millejoh** [comment]
Think I finally may be close to fixing this one. It appears the issue is due to not specifying the user agent when curl makes its request to the notebook server. This only seems to affect windows, for some reason. I'm experimenting with forcing the user agent to Mozilla/4.0. If all goes well with testing I will try to merge into the main branch ASAP.

In the meantime you can add `'("--user-agent" "Mozilla/4.0")` to the `request-curl-options` variable.

**Comment by millejoh** [comment]
Appveyor testing is not going well... The user-agent header seems to resolve authentication issues I have been seeing, but does not fix your JSON parsing error. I've stepped through code on both emacs and Python side - as far as I can tell emacs is doing everything right, but by the time it reaches the notebook server the JSON has lost some needed quotes (maybe curly braces as well).

Note this is not happening on my local machine and I have no idea what is different between my setup and the setup on appveyor.

**Comment by millejoh** [comment]
So it is looking like the JSON some Windows setups generates is not digestible by the jupyter notebook. Take, for example, when generating a new notebook. This happens by POST'ing the JSON string `{"type":"notebook"}` to the appropriate path on `\contents\api\{PATH}`.

In emacs, ein makes a call to `request`, which creates a temporary file that holds the json data for the POST. If you were to step through the code and find the temporary file the contents would look like:

```
{"type":"notebook"}
```

This works in Linux, Mac, and *some* Windows setups. On the Windows setups that don't work apparently the json needs to look like this:

```
{\"type\":\"notebook\"}
```

I have yet to determine what makes the difference between one working and the other not, but at this point I am about 90% sure this is the actual problem. So, yay?

**Comment by millejoh** [comment]
The plot thickens - originally the appveyor test build was using msys's curl, which is version 7.62.0 with release date 2018-10-31. This build fails on the functional tests, giving the invalid JSON errors.

If instead I install curl version 7.65.2 with release date 2019-07-17 using chocolatey the build succeeds on the functional tests.

My local machine has curl version 7.65.0 installed using choco and as noted previously, I have no problems running ein.

What is weird is that @c-vahl's version of curl is 7.65.1. The only other difference I can see is that his curl's architecture is `x86_64-w64-mingw32` and my and appveyor's curl is `x86_64-pc-win32`. It seems something is fishy in how mingw handles escaping of quote characters.

The official, most recently available version of [curl](https://curl.haxx.se/windows/) also seems to work fine with ein and the jupyter notebook.
